/**
 * Copyright 2018-2020 Streamlit Inc.
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *    http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
*/

syntax = "proto3";

message BlockPath {
  enum Container {
    UNUSED = 0;
    MAIN = 1;
    SIDEBAR = 2;
  }

  // Top level container of the block.
  Container container = 1;

  // A list with the indices of all ancestors of the block.
  repeated uint32 path = 2;
}
